src/rules/matchers.ts

```diff
@@ -260,6 +260,33 @@ export class RegexPathMatcher extends Serializable implements RequestMatcher {
 
 }
 
+export class RegexUrlMatcher extends Serializable implements RequestMatcher {
+    readonly type = 'regex-url';
+
+    readonly regexSource: string;
+    readonly regexFlags: string;
+
+    constructor(regex: RegExp) {
+        super();
+        this.regexSource = regex.source;
+        this.regexFlags = regex.flags;
+    }
+
+    matches(request: OngoingRequest) {
+        const absoluteUrl = normalizeUrl(request.url);
+        const urlPath = getPathFromAbsoluteUrl(absoluteUrl);
+
+        // Test the matcher against both the path alone & the full URL
```

Comment:
I think this comment and the `urlPath` variable are leftover from the other matcher, they can be deleted.

Reply:
Fixed